sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -157,6 +189,28 @@ public synchronized void close() {
     stop();
   }
 
+  private void sendChunks(final @NotNull IScopes scopes, final @NotNull SentryOptions options) {
+    try {
+      options
+          .getExecutorService()
+          .submit(
+              () -> {
+                final ArrayList<ProfileChunk> payloads = new ArrayList<>(payloadBuilders.size());
```

Comment:
any reason we don't store chunks (not builders) directly in [here](https://github.com/getsentry/sentry-java/pull/3711/files#diff-282328031af7b576dd167682cc966fe6852b8b0a6edaf27cc1c68427497d0ad2R162)? This seems like an unnecessary allocation of another list here

Reply:
this way we can call `build(SentryOptions)` when we send them
I did it because when we'll implement app start profiling we may not have the SDK initialized and the options when the profiler is running, so we'll wait to send the chunks as soon as we have them
But usually it should be a list with a single element
I'm using a different list so that we can clear the previous one in the synchronized block, without waiting for `scopes.captureProfileChunk(payload);`

Reply:
got it, makes sense!